internal/clientconn/conn.go

```diff
@@ -392,11 +390,42 @@ func (c *conn) handleOpMsg(ctx context.Context, msg *wire.OpMsg, cmd string) (*w
 }
 
 // Describe implements prometheus.Collector.
-func (l *conn) Describe(ch chan<- *prometheus.Desc) {
-	l.m.Describe(ch)
+func (c *conn) Describe(ch chan<- *prometheus.Desc) {
+	c.m.Describe(ch)
 }
 
 // Collect implements prometheus.Collector.
-func (l *conn) Collect(ch chan<- prometheus.Metric) {
-	l.m.Collect(ch)
+func (c *conn) Collect(ch chan<- prometheus.Metric) {
+	c.m.Collect(ch)
+}
+
+// logResponse logs response's header and body and returns the log level that was used.
+// If op code is not OP_MSG, it always logs as a debug.
+// For the OP_MSG code, the level depends on the type of error.
+// If there is no errors in the response, it will be logged as a debug.
+// If there is an error in the response, and connection is closed, it will be logged as an error.
+// If there is an error in the response, and connection is not closed, it will be logged as a warning.
+func (c *conn) logResponse(resHeader *wire.MsgHeader, resBody wire.MsgBody, closeConn bool) zapcore.Level {
+	level := zap.DebugLevel
+
+	if resHeader.OpCode == wire.OpCodeMsg {
+		doc, err := resBody.(*wire.OpMsg).Document()
```

Comment:
You could use `must.NotFail` here.

Reply:
@w84thesun I was thinking about it but I didn't get how :) Could you give me an example (e.g. where should I define a variable for `doc` etc)?

Reply:
Fixed